theano/gof/fg.py

```diff
@@ -402,7 +402,7 @@ def find_path_to(output_var, input_var):
                             assert path is not None
                             tr = getattr(r.tag, 'trace', [])
                             detailed_err_msg = ""
-                            if len(tr) > 0:
+                            if type(tr) is list and len(tr) > 0:
```

Comment:
Good question. I don't remember why (it's been months since I did this), but maybe the trace is not always a list but sometimes a string?

Anyway, I will change the code to use isinstance() instead.